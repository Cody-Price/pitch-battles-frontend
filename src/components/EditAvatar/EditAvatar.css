.edit-avatar-section {
  position: relative;
}

.avatar-editor-avatar-backer {
  grid-row: 1/3;
  margin: 2rem auto;
  position: relative;
  height: 80px;
  width: 80px;
  border-radius: 50%;
  background: url(../Game/images/offwhite-avatar-backer.png);
  background-size: cover;
  image-rendering: pixelated;
  border: 2px solid #4776bb;
  box-shadow: 1px 1px 1px#4776bb;
}

.avatar-editor-current-avatar-backer:hover {
  cursor: pointer;
}

.avatar-editor-current {
  margin: auto;
  height: 60px;
  width: 33px;
  position: absolute;
  left: 0;
  right: 0;
  bottom: 11px;
  transition: 0.2s;
}

.avatar-editor-current.one {
  background: url(../Game/pcs/char1/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-editor-current.two {
  background: url(../Game/pcs/char2/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-editor-current.three {
  background: url(../Game/pcs/char3/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-editor-current.four {
  background: url(../Game/pcs/char4/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.avatar-editor-current.five {
  background: url(../Game/pcs/char5/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 37px;
}

.avatar-editor-current.six {
  background: url(../Game/pcs/char6/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 37px;
}

.avatar-editor-current.nine {
  background: url(../Game/pcs/char9/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 36px;
  height: 60px;
}

.avatar-editor-current.ten {
  background: url(../Game/pcs/char10/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 42px;
  height: 60px;
}

.avatar-editor-current.seven {
  background: url(../Game/pcs/char7/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 42px;
  height: 62px;
}

.avatar-editor-current.eight {
  background: url(../Game/pcs/char8/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
  width: 43px;
  height: 62px;
}

.avatar-left-arrow:hover,
.avatar-right-arrow:hover {
  cursor: pointer;
}

.avatar-left-arrow:focus,
.avatar-right-arrow:focus {
  outline: none;
}

.avatar-left-arrow,
.avatar-right-arrow {
  z-index: 2000;

  height: 21px;
  width: 15px;
}

.avatar-left-arrow {
  position: absolute;
  background: url(../Onboarding/images/arrow-left.png);
  top: 80px;
  left: 150px;
  z-index: 100;
  background-size: cover;
  image-rendering: pixelated;
}

.avatar-right-arrow {
  position: absolute;
  background: url(../Onboarding/images/arrow-right.png);
  top: 80px;
  right: 150px;
  z-index: 100;
  background-size: cover;
  image-rendering: pixelated;
}

.avatar-left-arrow.false,
.avatar-right-arrow.false {
  display: none;
}

.edit-avatar-section button {
  grid-column: 1/3;
  width: 150px;
  height: 23px;
  font-family: "chronotype";
  font-size: 1.2rem;
  margin: 2rem auto;
  background: #6a9dd4;
  border: 0;
  display: block;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.7);
  padding-top: 4px;
}

.edit-avatar-section *:focus {
  outline: none;
}

.edit-avatar-section button:active {
  box-shadow: none;
}

.avatar-editor-label {
  text-align: center;
  margin-bottom: 2rem;
  font-size: 1.3rem;
}
