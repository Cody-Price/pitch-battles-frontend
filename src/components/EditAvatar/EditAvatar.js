import React, { Component } from "react";
import "./EditAvatar.css";
import avatars from "../../utilities/avatars";

class EditAvatar extends Component {
  constructor() {
    super();

    this.state = {
      avatar: 1,
      right: true,
      left: true
    };
  }
  componentDidMount() {
    this.checkAvatar();
  }

  checkAvatar = () => {
    if (this.props.user.attributes.avatar === 1) {
      this.setState({
        left: false,
        avatar: 1
      });
    } else if (this.props.user.attributes.avatar === 12) {
      this.setState({
        right: false,
        avatar: 12
      });
    } else {
      this.setState({
        right: true,
        left: true,
        avatar: this.props.user.attributes.avatar
      });
    }
  };

  handleRightClick = () => {
    if (this.state.avatar === 11) {
      this.setState({
        right: false,
        avatar: 12
      });
    } else {
      this.setState({
        left: true,
        avatar: this.state.avatar + 1
      });
    }
  };

  handleLeftClick = () => {
    if (this.state.avatar === 2) {
      this.setState({
        left: false,
        avatar: 1
      });
    } else {
      this.setState({
        right: true,
        avatar: this.state.avatar - 1
      });
    }
  };

  handleSubmit = () => {
    if (this.props.user.avatar === this.state.avatar) {
      return;
    }

    this.props.changeAvatar(this.state.avatar);
  };

  render() {
    return (
      <section className={`edit-avatar-section ${this.props.status}`}>
        <h3 className="avatar-editor-label">change your avatar</h3>
        <div className="edit-avatar-circle" />
        <div className="avatar-editor-avatar-backer">
          <div
            className={`avatar-editor-current ${avatars[this.state.avatar]}`}
          />
        </div>
        <div
          className={`avatar-left-arrow ${this.state.left}`}
          onClick={this.handleLeftClick}
        />
        <div
          className={`avatar-right-arrow ${this.state.right}`}
          onClick={this.handleRightClick}
        />
        <button onClick={this.handleSubmit} className="change-avatar-button">
          choose avatar
        </button>
      </section>
    );
  }
}

export default EditAvatar;
