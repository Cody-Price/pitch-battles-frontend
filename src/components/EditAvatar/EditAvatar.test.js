import React from "react";

import EditAvatar from "./EditAvatar";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("EditAvatar", () => {
  let wrapper;
  const mockUser = {
    attributes: { avatar: 2 }
  };
  const mockStatus = "true";
  let changeAvatar;

  beforeEach(() => {
    changeAvatar = jest.fn();
    wrapper = shallow(
      <EditAvatar user={mockUser} status={"true"} changeAvatar={changeAvatar} />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("checkAvatar", () => {
    it("should call setState if the user has avatar 1 on-load", () => {
      wrapper = shallow(
        <EditAvatar user={{ attributes: { avatar: 1 } }} status={mockStatus} />
      );

      const expected = {
        left: false,
        avatar: 1,
        right: true
      };

      expect(wrapper.state()).toEqual(expected);
    });

    it("should call setState if the user has avatar 12 on-load", () => {
      wrapper = shallow(
        <EditAvatar
          status={true}
          user={{ attributes: { avatar: 10 } }}
          status={mockStatus}
        />
      );

      const expected = {
        left: true,
        avatar: 12,
        right: false
      };
    });

    it("should call setState if the user has avatar 2-9 on-load", () => {
      const expected = {
        right: true,
        left: true,
        avatar: 2
      };

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("handleRightClick", () => {
    it("should call handleRightClick on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleRightClick");
      wrapper.instance().forceUpdate();

      wrapper.find(".avatar-right-arrow").simulate("click");

      expect(spy).toHaveBeenCalled();
    });

    it("should call setState if avatar is 13", () => {
      wrapper.setState({
        avatar: 13
      });

      const expected = {
        right: false,
        left: true,
        avatar: 14
      };

      wrapper.instance().handleRightClick();

      expect(wrapper.state()).toEqual(expected);
    });

    it("should call setState if avatar is any other value", () => {
      const expected = {
        left: true,
        avatar: 3,
        right: true
      };

      wrapper.instance().handleRightClick();

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("handleLeftClick", () => {
    it("should call handleLeftClick on click", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleLeftClick");
      wrapper.instance().forceUpdate();

      wrapper.find(".avatar-left-arrow").simulate("click");

      expect(spy).toHaveBeenCalled();
    });
    it("should call setState if avatar is 2", () => {
      wrapper.setState({
        avatar: 2
      });

      const expected = {
        right: true,
        left: false,
        avatar: 1
      };

      wrapper.instance().handleLeftClick();

      expect(wrapper.state()).toEqual(expected);
    });

    it("should call setState if avatar is any other value", () => {
      const expected = {
        left: true,
        avatar: 3,
        right: true
      };

      wrapper.setState({
        avatar: 4
      });

      wrapper.instance().handleLeftClick();

      expect(wrapper.state()).toEqual(expected);
    });
  });

  describe("handleSubmit", () => {
    it("should return if current avatar is the same as new avatar", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");

      wrapper.instance().handleSubmit();

      expect(spy).toReturn();
    });

    it("should call changeAvatar with the correct params", () => {
      wrapper.setState({
        avatar: 4
      });

      const expected = 4;

      wrapper.instance().handleSubmit();

      expect(changeAvatar).toHaveBeenCalledWith(expected);
    });
  });
});
