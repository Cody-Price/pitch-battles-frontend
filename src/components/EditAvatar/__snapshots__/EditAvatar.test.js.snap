// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAvatar should match the snapshot 1`] = `
<section
  className="edit-avatar-section true"
>
  <div
    className="avatar-editor-current two"
  />
  <div
    className="avatar-left-arrow true"
    onClick={[Function]}
  />
  <div
    className="avatar-right-arrow true"
    onClick={[Function]}
  />
  <button
    className="change-avatar-button"
  >
    choose avatar
  </button>
</section>
`;
